client/state/data-layer/wpcom/jetpack/connection/owner/index.js

```diff
@@ -0,0 +1,61 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { JETPACK_CONNECTION_CHANGE_OWNER } from 'state/action-types';
+import { requestJetpackUserConnectionData } from 'state/jetpack/connection/actions';
+
+const changeConnectionOwner = action =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'POST',
+			path: '/jetpack-blogs/' + action.siteId + '/rest-api/',
+			body: {
+				path: '/jetpack/v4/connection/owner/',
+				body: JSON.stringify( { owner: action.newOwnerDotorgId } ),
+				json: true,
+			},
+		},
+		action
+	);
+
+const handleSuccess = ( { newOwnerWpcomDisplayName, siteId } ) => {
+	return dispatch => {
+		dispatch(
+			successNotice(
+				translate( 'Connection owner changed to %(user)s.', {
+					args: { user: newOwnerWpcomDisplayName },
+				} )
+			)
+		);
+		dispatch( requestJetpackUserConnectionData( siteId ) );
+	};
+};
```

Comment:
I think we can just return an array of actions like we do here:

https://github.com/Automattic/wp-calypso/blob/1439f80dc4884244a92f92d07da5a8ee5c38e924/client/state/data-layer/wpcom/me/two-step/application-passwords/new/index.js#L52-L55